src/Microsoft.AspNet.Server.Kestrel/Http/FrameRequestStream.cs

```diff
@@ -94,7 +94,7 @@ public Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationTok
                 {
                     tcs2.SetResult(task2.Result);
                 }
-            }, tcs);
+            }, tcs, cancellationToken);
             return tcs.Task;
```

Comment:
I'm not against `private` here - but in that case, it should just be moved into the `#if NET451` altogether IMO - is anyone external possibly depending on this?